File: Proyecto/apiRest_users_kotlin_jpc/kotlin-users-service-api/src/main/kotlin/com/puj/admincenter/dto/users/UserDto.kt
package com.puj.admincenter.dto.users

import com.puj.admincenter.domain.users.User

data class UserDto(
    val email: String,
    val id: Int,
    val name: String,
    val password:String,
    var username: String
) {
    companion object {
        fun convert(user: User): UserDto {
            val dto = UserDto(
                email= user.email,
                id = user.id,
                name = user.name,
                password= user.password,
                username = user.username
            )
            return dto
        }
    }
}

File: Proyecto/WebDev/WebDev.Services/WebDev.Services/Entities/UserDto.cs
namespace WebDev.Services.Entities
{
    public class UserDto
    {
        public string email { get; set; }
        public string name { get; set; }
        public string username { get; set; }
        public string password { get; set; }
        public int id { get; set; }

        private UserDto()
        {
        }

        public static UserDto Build(string email, string name, string username, string password, int id)
        {
            return new UserDto
            {
                email = email,
                name = name,
                username = username,
                password = password,
                id = id
            };
        }
    }
}